signature/steps.go

```diff
@@ -12,13 +12,14 @@ var errSigningRefusedUnknownStepType = errors.New("refusing to sign pipeline con
 
 // SignSteps adds signatures to each command step (and recursively to any command steps that are within group steps).
 // The steps are mutated directly, so an error part-way through may leave some steps un-signed.
-func SignSteps(ctx context.Context, s pipeline.Steps, key Key, repoURL string, opts ...Option) error {
+func SignSteps(ctx context.Context, s pipeline.Steps, key Key, repoURL string, env map[string]string, opts ...Option) error {
```

Comment:
It doesn't feel that this change would need to remove replace the `WithEnv`? Would it be better to keep move it to this file, and let its `apply` to modify the `OuterEnv`?